From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: risbun <risbun.inc@gmail.com>
Date: Mon, 18 Jul 2022 04:51:00 +0200
Subject: [PATCH] Ender dragon knockback fix


diff --git a/src/main/java/net/minecraft/server/Explosion.java b/src/main/java/net/minecraft/server/Explosion.java
index e408ec8c645e9ceb278a13a5b0de8ecf57f3929e..a9208d585279584907376acce6963eeaa4e5c1df 100644
--- a/src/main/java/net/minecraft/server/Explosion.java
+++ b/src/main/java/net/minecraft/server/Explosion.java
@@ -182,7 +182,7 @@ public class Explosion {
                         entity.forceExplosionKnockback = false;
                         boolean wasDamaged = entity.damageEntity(this.b(), (float) ((int) ((d13 * d13 + d13) / 2.0D * 7.0D * (double) f2 + 1.0D)));
                         CraftEventFactory.entityDamage = null;
-                        if (!wasDamaged && !(entity instanceof EntityTNTPrimed || entity instanceof EntityFallingBlock) && !entity.forceExplosionKnockback) {
+                        if (!wasDamaged && !(entity instanceof EntityTNTPrimed || entity instanceof EntityFallingBlock || entity instanceof EntityEnderDragon) && !entity.forceExplosionKnockback) {
                             continue;
                         }
                         // CraftBukkit end
